ksql-serde/src/main/java/io/confluent/ksql/serde/connect/ConnectFormat.java

```diff
@@ -0,0 +1,142 @@
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.serde.connect;
+
+import com.google.common.annotations.VisibleForTesting;
+import io.confluent.kafka.schemaregistry.ParsedSchema;
+import io.confluent.ksql.name.ColumnName;
+import io.confluent.ksql.schema.ksql.SchemaConverters;
+import io.confluent.ksql.schema.ksql.SimpleColumn;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import io.confluent.ksql.serde.Format;
+import io.confluent.ksql.util.DecimalUtil;
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import org.apache.kafka.connect.data.Field;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+
+public abstract class ConnectFormat implements Format {
+
+  private final Function<Schema, Schema> toKsqlTransformer;
+
+  public ConnectFormat() {
+    this(new ConnectSchemaTranslator()::toKsqlSchema);
+  }
+
+  @VisibleForTesting
+  ConnectFormat(final Function<Schema, Schema> toKsqlTransformer) {
+    this.toKsqlTransformer = Objects.requireNonNull(toKsqlTransformer, "toKsqlTransformer");
+  }
+
+  @Override
+  public boolean supportsSchemaInference() {
+    return true;
+  }
+
+  @Override
+  public List<SimpleColumn> toColumns(final ParsedSchema schema) {
+    final Schema connectSchema = toKsqlTransformer.apply(toConnectSchema(schema));
+
+    return connectSchema.fields().stream()
+        .map(ConnectFormat::toColumn)
+        .collect(Collectors.toList());
+  }
+
+  public ParsedSchema toParsedSchema(final List<SimpleColumn> columns) {
+    final SchemaBuilder schemaBuilder = SchemaBuilder.struct();
+    columns.forEach(col -> schemaBuilder.field(
+        col.ref().name(),
+        SchemaConverters.sqlToConnectConverter().toConnectSchema(col.type()))
+    );
+
+    final Schema connectSchema = ensureNamed(schemaBuilder.build());
+    return fromConnectSchema(connectSchema);
+  }
+
+  protected abstract Schema toConnectSchema(ParsedSchema schema);
+
+  protected abstract ParsedSchema fromConnectSchema(Schema schema);
+
+  private static SimpleColumn toColumn(final Field field) {
+    final ColumnName name = ColumnName.of(field.name());
+    final SqlType type = SchemaConverters.connectToSqlConverter().toSqlType(field.schema());
+    return new ConnectColumn(name, type);
+  }
+
+  private static Schema ensureNamed(final Schema schema) {
```

Comment:
previously we weren't doing this, why do we need to do it here?

Reply:
We were. This was a cut&paste from `TestCaseBuilderUtil.addNames()`